pkg/integrations/prometheus/promql/metricsSearchHandler.go

```diff
@@ -403,35 +403,56 @@ func ProcessGetMetricTimeSeriesRequest(ctx *fasthttp.RequestCtx, myid uint64) {
 		utils.SendError(ctx, "empty json body received", "", nil)
 		return
 	}
+	qid := rutils.GetNextQid()
 
 	start, end, queries, formulas, errorLog, err := parseMetricTimeSeriesRequest(rawJSON)
 	if err != nil {
 		utils.SendError(ctx, err.Error(), errorLog, err)
 		return
 	}
 
-	log.Debugf("ProcessGetMetricTimeSeriesRequest: start=%v, end=%v, queries=%v, formulas=%v", start, end, queries, formulas)
-	// TODO: Integrate functionality to get metric time series and remove dummy response
-	dummyResponse := `{
-		"aggStats": {
-			"testmetric0": {
-				"2024-05-01T12:59": 0,
-				"2024-05-01T13:00": 0,
-			},
-			"testmetric1": {
-				"2024-05-01T12:59": 0,
-				"2024-05-01T13:00": 0,
-			},
+	log.Debugf("ProcessGetMetricTimeSeriesRequest: Need to Parse: formulas=%v", formulas)
+
+	metricQueriesList := make([]*structs.MetricsQuery, 0)
+	var timeRange *dtu.MetricsTimeRange
+	hashList := make([]uint64, 0)
+
+	// Todo:
+	// The queryFormulas var should contain the parsed formulas.
+	// Modify the below flow to parse the individual queries, execute the queries and then apply the formulas.
+	var queryFormulas []structs.QueryArithmetic
+	for _, query := range queries {
+		searchText, start, end, interval := parseSearchTextForRangeSelection(fmt.Sprintf("%v", query["query"]), start, end)
+		metricQueryRequest, pqlQuerytype, queryArithmetic, err := convertPqlToMetricsQuery(searchText, start, end, myid)
+		if err != nil {
+			utils.SendError(ctx, "Error parsing metrics query", fmt.Sprintf("Metrics Query: %+v", searchText), err)
+			return
 		}
+		// The size of the metricQueryRequest might always be 1.
+		for _, metricQuery := range metricQueryRequest {
+			metricQuery.MetricsQuery.PqlQueryType = pqlQuerytype
+			metricQuery.MetricsQuery.Interval = interval
+			hashList = append(hashList, metricQuery.MetricsQuery.HashedMName)
+			metricQueriesList = append(metricQueriesList, &metricQuery.MetricsQuery)
+			segment.LogMetricsQuery("PromQL metrics query parser", &metricQuery, qid)
+			timeRange = &metricQuery.TimeRange
+		}
+		queryFormulas = queryArithmetic
 	}
-	`
-	ctx.SetBody([]byte(dummyResponse))
+
+	res := segment.ExecuteMultipleMetricsQuery(hashList, metricQueriesList, queryFormulas, timeRange, qid)
+	mQResponse, err := res.GetResultsPromQlForUi(metricQueriesList[0], metricQueriesList[0].PqlQueryType, start, end, metricQueriesList[0].Interval)
+	if err != nil {
+		log.Errorf("ExecuteAsyncQuery: Error getting results! %+v", err)
```

Comment:
nit: func name should be `ProcessGetMetricTimeSeriesRequest` and log line should be `...failed to get results ...`

Reply:
we should also log the qid for all error messages in this functions